tests/testthat/test-pareto_smooth.R

```diff
@@ -0,0 +1,194 @@
+test_that("pareto_khat returns expected reasonable values", {
+  tau <- extract_variable_matrix(example_draws(), "tau")
+
+  pk <- pareto_khat(tau)
+  expect_true(names(pk) == "khat")
+
+})
+
+test_that("pareto_khat handles tail argument", {
+
+  # as tau is bounded (0, Inf) the left pareto k should be lower than
+  # right
+  tau <- extract_variable_matrix(example_draws(), "tau")
+  pkl <- pareto_khat(tau, tail = "left")
+  pkr <- pareto_khat(tau, tail = "right")
+  pkb <- pareto_khat(tau)
+  expect_true(pkl$khat < pkr$khat)
+  expect_equal(pkr$khat, pkb$khat)
+})
+
+test_that("pareto_khat handles ndraws_tail argument", {
+
+  tau <- extract_variable_matrix(example_draws(), "tau")
+  pk10 <- pareto_khat(tau, tail = "right", ndraws_tail = 10)
+  pk25 <- pareto_khat(tau, tail = "right", ndraws_tail = 25)
+  expect_true(pk10$khat > pk25$khat)
+
+  expect_warning(pareto_khat(tau, tail = "both", ndraws_tail = 201),
+                 "Number of tail draws cannot be more than half ",
+              "the length of the draws if both tails are fit, ",
```

Comment:
the total number of draws if both tails are fit

Reply:
Changed